### ISSUE TITLE
Feature request. Batch Merge in worksheet class

### ISSUE THREAD

**Issue by MildogMiller** [burnash/gspread#1473]
**Is your feature request related to a problem? Please describe.**
I am using gspread to create quite a large spreadsheet in a particular format. One of the things that is very present in the worksheet are merged cells. The `batch_update`, `batch_format`, and `batch_get` have been invaluable to me. However, there is no way from the worksheet side (that i can see) to do a batch merge.

I understand that logically merging cells should be done before you insert data. One thing that i find myself doing is having a long list of `merge_cells` at the beginning of my script, which sets up all the merges. However, each makes an API request.

**Describe the solution you'd like**
I would like a `batch_merge` method defined in the `Worksheet class` that takes in a list of ranges and uses the `batch_update` method of the Google Sheets API, resulting in only 1 API request. It would most likely need to append multiple `mergeCells` to a request body.

**Describe alternatives you've considered**
Currently one of the ways i get around it by making the request myself and then using the `Speadsheet.batch_update` method. However, it would be better if i could do it through some sort of abstraction and keep everything in the `Worksheet` class

**Additional context**
N/A

I am more than happy to attempt to write the method on my own and create a pull request. I understand it is a bit of a niche issue

**Comment by MildogMiller** [comment]
Thank you so much @lavigne958 for your speedy reply on the issue. Yes that is indeed what i currently do and it does work perfectly.

I should say the reason i requested the feature is more from an abstraction side. One of those nice to haves. It is definitely a feature that is low priority.

Thanks again!

**Comment by muddi900** [comment]
> When we get a chance we'll add this feature.
> 
> for the new feature I propose we make this way:
> 
> ```python
> spreadsheet.batch_merge(merges: Dict[str, MergeType])
> ```
> 
> The above type is to be adjusted after some test.

@lavigne958 wouldn't it be better to have this method on the `Worksheet` level. Also, the type of `merges` params should be `List[Dict]`.

**Comment by lavigne958** [comment]
> > When we get a chance we'll add this feature.
> > for the new feature I propose we make this way:
> > ```python
> > spreadsheet.batch_merge(merges: Dict[str, MergeType])
> > ```
> > The above type is to be adjusted after some test.
> 
> @lavigne958 wouldn't it be better to have this method on the `Worksheet` level. Also, the type of `merges` params should be `List[Dict]`.

yes, it would be better to have it at the worksheet level. 
no it should be a `Dict`, each key of the dict will be a range, then each value for that range will be a merge type.

the pseudo code should like this:

```
requests: empty list
for each pair (key, value) in merges:
    full_range = a1_range_to_grid_range(key, sheet_id)
    append to request: {"MergeCells": {"range": full_range, "mergeType": value}}
end for loop

send batch_update(requests)
```

unless I miss something ?
